### DISCUSSION TITLE
Landed a Job as a Blockchain Developer!

### DISCUSSION THREAD

**Discussion by Niferu** [Cyfrin/foundry-full-course-cu#1859] [Show and tell]
Hello Folks,

I thought it would be nice to write this post to encourage everyone who works hard and to assure them that it is possible to get a job as a programmer in the blockchain industry (even now, when the market is very rough). As the title suggests, I have finally landed my first job in this amazing industry without previous experience as a developer! I'm also self-taught; I did not study programming or anything similar formally. I started learning with Patrick's course for Solidity and Python. Then, I finished the JS and Foundry versions, completing all the challenges and exercises actively helping community and even improving courses repositiories.

Regarding searching for job, it took me a veeeeery long time (compared to my regular job searches where one application usually equaled a new job for me). I was programming every single day for a minimum of 2 hours (usually much longer because I truly love it). I quickly began to create my own projects from start to end, which evolved into more advanced projects for a few Chainlink Hackathons or for my friends, and I learned tons, including diving into frontend development!

I sent a lot of applications (over 200-300, hard to tell exactly), and I mostly received only a few responses from recruiters who didn't even know how to code. After some time when not getting any responses for blockchain positions I started to applying for regular programmer jobs (it was everything really backend, frontend, fullstack literally anything that matched my stack a bit), but it failed as well as all of my projects were web3, so it did not worked for regular positions and I was not as good lets say in frontend as I was in Solidity for example. After continuing to learn and improve my CV while sending more applications, I received only three invitations for technical interviews (with breaks of about 1-2 months between each invitation). I was scared as hell because we all know how hard and stressful technical interviews can be, but to my surprise, I did really well. The first interview failed because of my long notice period, the second one failed because the founders of the project didn't want a programmer without previous experience (even though the programmers who interviewed me wanted to give me the job). Finally, the third interview went very smoothly as well, but this time successfully and I beat over 70 other candidates for the position, landing my first job as a blockchain engineer!

**Advice for Everyone Who Keeps Trying:**
1. **Build Cool Projects:** Focus on creating amazing projects. It's better to concentrate on one outstanding project than on several common ones.

2. **Pursue Your Interests:** Focus only on roles that genuinely interest you. If you like frontend development, go for it, but don't mix it with other areas if you are new. Employers prefer specialists over generalists who know a little about everything but are not experts in any single thing.

3. **Enhance Your Profile:** Keep refining your CV, create a portfolio website, update your GitHub profile README, talk to other developers, participate in hackathons, and contribute as often as you can. Show your activity on GitHub to demonstrate your ongoing engagement and skills.

Thank you, @PatrickAlphaC!!!
I hope that you might consider filling the gap with courses for Solana (Anchor framework) blockchain programming I would love to dive into it more :) You are amazing and keep doing great stuff!

I hope this motivates everyone who works as hard as I do to reach their goals!

**Comment by EngrPips** [comment]
I am highly motivated, and I hope to achieve my goals too. Keep crushing it, @Nlferu

**Reply by sharonphiliplima** [reply]
you have always been helpful to me in the discussions. sending prayers!

**Comment by urosognjenovic** [comment]
I'm glad to hear this! I wish you all the best of luck!

**Comment by PatrickAlphaC** [comment]
Thank you so much for sharing your progress here, this is MASSIVE!! 

Can we use this as a testimonial for the course?

**Reply by Niferu** [reply]
@PatrickAlphaC

Sure thing Patrick, you can of course! And thanks a lot again!

**Comment by zubyoha** [comment]
Thank you so much for sharing, this has really motivated me to keep pushing. @Nlferu

**Reply by EngrPips** [reply]
Indeed, it motivated me a lot, too!

**Comment by a1111198** [comment]
Thanks @Nlferu for sharing this, After reading it I felt that I am on the right track it is just that process is taking its time. I  also missed my 2 technical interviews one at nethermind and another is at valory. But yes I am still applying and learning.
Also @PatrickAlphaC could you plese share some resources where you mentioned the ways to apply and platforms to get the jobs ( may be we are missing something in the process of applying). Also would love to have a video on this toipic by you.

**Comment by CH-coding-fire** [comment]
This is LEGENDARY!!! I can't imagine a person without programming background can get a smart contract job. I have 2 YOE full-stack but I still really doubt if I can get a job, lets PUSH harder.

**Reply by CH-coding-fire** [reply]
Btw, it would be even more helpful if you can disclose the package you received, such as salary, work-from-home, hours, career prospect. Because it can give a idea to people what to expect

**Reply by Niferu** [reply]
Hi @CH-coding-fire 

Everything you mentioned is highly dependent on the country you live in. I had no chance when applying for global positions because there were ten times more developers trying to get the same job. For me, with a financial education and no professional programming experience, competing with such people was virtually impossible.

**Advice:**
For anyone with a similar educational background and no experience, I highly recommend searching for a job in your own country (there will be far fewer candidates). As I mentioned, I beat around 70 other candidates on LinkedIn alone, but who knows how many more there were from other websites.

**Expectations:**
I was earning a very good salary in my financial position, but I was so motivated to change jobs that I was even willing to accept half of my current salary to get a new job. The company that hired me is a startup from my country. They were so impressed by my skills, after beating all those candidates, that I received much more compensation than I currently had. However, as I mentioned, money doesn't matter to me; I just want to do what I really like. The offer I received is also fully remote (I was free to choose), and it is a standard 8-hour workday. Regarding career prospects, as I said, it is a startup, so if everything goes well, it will be amazingly beneficial for me because I will be building it from scratch. I'm not at a junior level in this job but more likely mid-senior. Even if it somehow fails, which I really doubt, I will have my first professional experience on board, making it much easier for me to get another job.

**Comment by EngrPips** [comment]
I mean this man reminds us that we can achieve whatever we set out to achieve

**Comment by Robschuh** [comment]
Thanks for sharing your experience, @Nlferu. I agree with you that @PatrickAlphaC's courses are amazing. He is doing a great job in the Web3.0 community. As an experienced backend developer (in my case, I am a PHP engineer), I find Patrick's courses to be the best training for advancing my transition to Web 3.0. I am quite far from being a good Solidity engineer, but I strongly believe that one day I will make it! As I like to say to people who ask me how to improve their programming skills, I compare a programmer to a swimmer. You can learn how to swim pretty well, but the difference between an Olympic swimmer and a good or average one is the training. Swim, swim, and swim every day, and you will succeed. 
Good job!!

**Reply by EngrPips** [reply]
Well said, sir!

**Comment by tamermint** [comment]
Thanks mate, really needed to read this and sending best wishes!  I am about to finish my Advanced Foundry but I feel like I don't know anything :( @PatrickAlphaC what's the best way for someone with no professional programming experience to have that fluency to start coding. When I start building, I hit a brick wall - like a plateau

**Reply by EngrPips** [reply]
when you hit a brick wall do research and find out the solution to that brick wall and continue coding and you will run into another brick wall and you will repeat what you did to solve the previous brick wall and that is how you will continue living your life till you die or retire as a programmer.

**Reply by Niferu** [reply]
Exactly as  @EngrPips said! Programming is a continuous battle between you and problems, new technologies, updated libraries, etc. That's just how it is—so you either love it or hate it. Fastest way to learn is to build your own project from scratch otherwise it was impossible for me either to know everything from course. I understood most of course things just after doing my projects and implementing similar solutions, just give yourself time

**Reply by tamermint** [reply]
Thanks @EngrPips and @Nlferu for your words - hoping to have my own startup soon. I have too many ideas in my head and but not enough implementation knowledge :)

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.

**Comment by EthanWalker10** [comment]
What a fabulous experience I first read your previous [Q&A](https://github.com/Cyfrin/foundry-full-course-cu/discussions/503) and then I got this. I'm a recent graduate who wanna engage into the awesome field. You made a cool thing and you will passionately continue it, which really inspires me and gives me a lot of confidence.   
  
Best wishes to you!

**Reply by EngrPips** [reply]
Interesting.

**Comment by darmeey** [comment]
> Hello Folks,
> 
> I thought it would be nice to write this post to encourage everyone who works hard and to assure them that it is possible to get a job as a programmer in the blockchain industry (even now, when the market is very rough). As the title suggests, I have finally landed my first job in this amazing industry without previous experience as a developer! I'm also self-taught; I did not study programming or anything similar formally. I started learning with Patrick's course for Solidity and Python. Then, I finished the JS and Foundry versions, completing all the challenges and exercises actively helping community and even improving courses repositiories.
> 
> Regarding searching for job, it took me a veeeeery long time (compared to my regular job searches where one application usually equaled a new job for me). I was programming every single day for a minimum of 2 hours (usually much longer because I truly love it). I quickly began to create my own projects from start to end, which evolved into more advanced projects for a few Chainlink Hackathons or for my friends, and I learned tons, including diving into frontend development!
> 
> I sent a lot of applications (over 200-300, hard to tell exactly), and I mostly received only a few responses from recruiters who didn't even know how to code. After some time when not getting any responses for blockchain positions I started to applying for regular programmer jobs (it was everything really backend, frontend, fullstack literally anything that matched my stack a bit), but it failed as well as all of my projects were web3, so it did not worked for regular positions and I was not as good lets say in frontend as I was in Solidity for example. After continuing to learn and improve my CV while sending more applications, I received only three invitations for technical interviews (with breaks of about 1-2 months between each invitation). I was scared as hell because we all know how hard and stressful technical interviews can be, but to my surprise, I did really well. The first interview failed because of my long notice period, the second one failed because the founders of the project didn't want a programmer without previous experience (even though the programmers who interviewed me wanted to give me the job). Finally, the third interview went very smoothly as well, but this time successfully and I beat over 70 other candidates for the position, landing my first job as a blockchain engineer!
> 
> **Advice for Everyone Who Keeps Trying:**
> 
> 1. **Build Cool Projects:** Focus on creating amazing projects. It's better to concentrate on one outstanding project than on several common ones.
> 2. **Pursue Your Interests:** Focus only on roles that genuinely interest you. If you like frontend development, go for it, but don't mix it with other areas if you are new. Employers prefer specialists over generalists who know a little about everything but are not experts in any single thing.
> 3. **Enhance Your Profile:** Keep refining your CV, create a portfolio website, update your GitHub profile README, talk to other developers, participate in hackathons, and contribute as often as you can. Show your activity on GitHub to demonstrate your ongoing engagement and skills.
> 
> Thank you, @PatrickAlphaC!!! I hope that you might consider filling the gap with courses for Solana (Anchor framework) blockchain programming I would love to dive into it more :) You are amazing and keep doing great stuff!
> 
> I hope this motivates everyone who works as hard as I do to reach their goals!

This is very motivating. Thank you for sharing.

**Comment by darmeey** [comment]
very motivating. Thank you for sharing
Happy for you

**Reply by EngrPips** [reply]
Indeed.

**Comment by Napcalx** [comment]
Congratulations on this amazing feat,.

**Comment by Ifra001** [comment]
WOW this is amazing!! This just proves that hard work won't fail you and that you just have to keep trying and trying. This motivated me to be consistent with my coding journey to! Thank you for sharing this! 
Also Congrats to you!!

**Reply by EngrPips** [reply]
Success is inevitable provided intensity and consistency remains very high.

**Comment by laurarodg** [comment]
Really want to say thank you for sharing this. I'm into web3 and even if it takes a long journey (probably will) i'm willing to give it a 100%. Hopefully i'll be back here in a few years to inspire others.

**Reply by Niferu** [reply]
@laurarodg 

If this is what you truly love to do, I can assure you it’s absolutely worth it! I enjoyed my previous job in finance, but now… damn, being a programmer and working on cool stuff is just awesome!

**Comment by KMean** [comment]
@Nlferu Wow that's really inspiring. Thanks for sharing!

**Reply by EngrPips** [reply]
Indeed.

**Comment by OtowoSamuel** [comment]
It's scary the time it took.
But it's just to keep looking ahead

**Reply by EngrPips** [reply]
You need to be exceptionally good and dedicated to stand a chance at all.
